stdlib/src/collections/list.mojo

```diff
@@ -335,6 +335,61 @@ struct List[T: CollectionElement](CollectionElement, Sized):
             earlier_idx += 1
             later_idx -= 1
 
+    @always_inline
+    fn index(self, owned value: T, start: Int = 0) raises -> Int:
+        """Returns the index of the first occurrence of a value in a list; raises ValueError if not found."""
+        var normalized_start = self.size + start if start < 0 else start
+        var normalized_end = self.size
+    
+        if normalized_start >= self.size: raise "Given 'start' parameter is out of range"
```

Comment:
**Suggestion** It may be good to include `normalized_start` as well as `len(self)`, i.e. the list size in the error message raised.  Ditto in the overload below.

Reply:
Great idea for improving error messages in Mojo, similar to the enhancements we've observed in CPython over recent years. I will implement this as well. Thanks

Reply:
One quick question. In Python, if you give out-of-range start/end values, it does not raise any error concerning that bad given range. It just raises a value error in any case that it could not find the value. So, in Python, the code below works but in my implementation, it does not work:

```python
x = [10, 20, 30]
x.index(10, 0, 500) # Search for value; If does not exists, raise a Value Error. Don't care if start/end is out of range
```

So do you think we should keep this raising as it is? Or because of this incompatibility, it's better to drop it off?

I suddenly noticed about this so I personally prefer to have the same behavior in Mojo as well.

Reply:
I fixed it to use an appropriate approach. Neither losing clarity of the error messages you desired nor making incompatibility with Python's API.